src/WebSocketConnection.php

```diff
@@ -32,22 +32,17 @@ function (Frame $frame) {
                         $this->stream->write((new Frame($frame->getPayload(), true, Frame::OP_PONG))->getContents());
                         return;
                     case Frame::OP_CLOSE:
-                        $closeCode = 1000;
-                        if ($frame->getPayloadLength() >= 2) {
-                            list($closeCode) = array_merge(unpack('n*', substr($frame->getPayload(), 0, 2)));
-                        }
-
-                        $this->stream->end($frame->getContents());
+                        $closeCode = unpack('n*', substr($frame->getPayload(), 0, 2));
+                        $closeCode = reset($closeCode) ?: 1000;
+                        $reason = '';
 
-                        if ($closeCode >= 2000) {
-                            // emit close code as error
-                            $exception = new \Exception('WebSocket closed with code ' . $closeCode);
-                            $this->emit('error', [$exception, $this]);
-                            return;
+                        if ($frame->getPayloadLength() > 2) {
+                            $reason = substr($frame->getPayload(), 2);
                         }
 
-                        $this->emit('close', [$closeCode, $this]);
+                        $this->stream->end($frame->getContents());
 
+                        $this->emit('close', [$closeCode, $reason, $this]);
```

Comment:
This close code emission is a breaking change. I do like sending the reason back, is there a way to do this without breaking this?

Reply:
We could just put the reason as the 3rd element and keep `$this` as the 2nd element. Existing code would continue to work then.

Reply:
That sounds good to me. I you want to make that change I can get this merged.

Reply:
@mbonneau Changes are pushed.